### ISSUE TITLE
Improve Photoshop export script

### ISSUE THREAD

**Issue by Shiu** [EsotericSoftware/spine-editor#202]
1. Update the photoshop script to get a native interface look. These are called Panels in Photoshop. For some examples of what is possible take a look at http://www.pspanels.com/ 

2. We currently have the option of exporting groups as skins. It would be great if we could also export groups as slot images. Maybe we can find a way to have both options. We could use naming conventions for the group types. 

3. Better individual layer export. Panels open up the possibility of drag and drop. For example, drag a layer onto a defined panel area to export it with pre-defined settings.

4. Handle all layer types. In some cases, this means ignoring a layer type, in others it would be rasterizing before export. To keep it simple we could just display a warning for layers that cannot be exported.

5. Partial skeleton export. We already allow exporting a JSON file, but this exports a full skeleton. Maybe we can find a way to allow exporting a partial skeleton that can then be imported onto an existing skeleton in a simple way.

These are all just suggestions. Keep the ideas coming.

Suggestion from Geddski #169

The script doesn't handle duplicate layer names well either. Say I have a model that has 2 arms (pretty common) with a layer tree like this:

```
zombie
  |--left arm
      |--hand
  |--right arm
      |--hand
```
once imported into Spine I would be left with only a single hand since the layers have the same name. The workaround is to make sure every single layer has a unique name which is tedious and frustrating to work with. Spine could detect the duplicate name and just append a number to the end.

Or it could prepend the layer name.

**Comment by NathanSweet** [comment]
Ensure file extensions aren't lost:
http://esotericsoftware.com/forum/Missing-Unread-Image-Files-from-Photoshop-8842

**Comment by NathanSweet** [comment]
1) The script already uses native UI controls. @Shiu pspanels.com doesn't exist. I'm not convinced the script needs to be visible all the time. At least with the current functionality, it makes sense to run only when needed (can use a hotkey to trigger an action).

2) Done.

3) We have naming conventions to control per layer settings. Ideas for more settings are welcome.

4) All layers are rasterized. @Shiu Please file issues for PSDs the script doesn't support.

5) The plan is to allow `Import Data` to bring attachments into an existing project. If an attachment already exists, we can ignore or replace it.

6) Currently for duplicate layer names we use only the first and ignore the rest. This can be useful when an alt pose duplicates some layers. It is bad it's a silent error if that is not what the user wants. The options are: ignore, error, or rename. We could have an `Ignore duplicate layer names` checkbox, but that would not provide the option to automatically rename. Users should understand that their layer names are being used for both image file names and slot names, so should not have duplicate layer names.

**Comment by pharan** [comment]
http://esotericsoftware.com/forum/problems-with-latest-photoshop-script-9284

**Comment by pharan** [comment]
Confirmed the bug reported by the user.

Layer Setup:
![nvidia share_2017-09-15_11-49-29](https://user-images.githubusercontent.com/1911082/30465831-2f72ee9e-9a0c-11e7-83ea-383acc7de3f6.png)
![nvidia share_2017-09-15_11-49-33](https://user-images.githubusercontent.com/1911082/30465838-399a6cb2-9a0c-11e7-9967-3eabd5a7cf31.png)

Exported images:
![nvidia share_2017-09-15_11-49-43](https://user-images.githubusercontent.com/1911082/30465844-49173756-9a0c-11e7-9f4c-e05690ff6d84.png)

**Comment by NathanSweet** [comment]
Fixed in EsotericSoftware/spine-scripts@c0b9822b8973666d248c333a48d3d38cd29548d4
